src/app/components/gv-documentation/gv-documentation.component.ts

```diff
@@ -234,7 +238,7 @@ export class GvDocumentationComponent implements AfterViewInit {
 
   private getFirstPage(pages: any[], pageId?: string) {
     for (const page of pages) {
-      if (this.isAsciiDoc(page) || this.isSwagger(page) || this.isMarkdown(page)) {
+      if (this.isAsciiDoc(page) || this.isAsyncApi(page) || this.isSwagger(page) || this.isMarkdown(page)) {
```

Comment:
The whole methods can be simplified with something like:
```ts

  private getFirstPage(pages: Page[], pageId?: string): Page | undefined {
    return pages
      .filter((page) => this.isAsciiDoc(page) || this.isAsyncApi(page) || this.isSwagger(page) || this.isMarkdown(page))
      .find((page) =>
        // Check pageId if input is defined otherwise fallback on the first page
        pageId ? pageId === page.id : true,
      );
  }
```

Reply:
yes, i prefer this stream syntax.
I refactored this function like this, and added unit tests to ensure non regression of the new syntax.